pkg/anchor/policy/resthandler/configurator.go

```diff
@@ -0,0 +1,135 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package resthandler
+
+import (
+	"io/ioutil"
+	"net/http"
+	"net/url"
+
+	"github.com/hyperledger/aries-framework-go/spi/storage"
+	"github.com/trustbloc/edge-core/pkg/log"
+	"github.com/trustbloc/sidetree-core-go/pkg/restapi/common"
+
+	"github.com/trustbloc/orb/pkg/activitypub/resthandler"
+	"github.com/trustbloc/orb/pkg/activitypub/store/spi"
+	"github.com/trustbloc/orb/pkg/anchor/policy"
+	"github.com/trustbloc/orb/pkg/anchor/policy/config"
+)
+
+const endpoint = "/policy"
+
+const (
+	unauthorizedResponse        = "Unauthorized."
+	badRequestResponse          = "Bad Request."
+	internalServerErrorResponse = "Internal Server Error."
+)
+
+var logger = log.New("policy-rest-handler")
+
+type signatureVerifier interface {
+	VerifyRequest(req *http.Request) (bool, *url.URL, error)
+}
+
+// PolicyConfigurator updates witness policy in config store.
+type PolicyConfigurator struct {
+	*resthandler.AuthHandler
+
+	VerifyActorInSignature bool
+	configStore            storage.Store
+}
+
+// Path returns the HTTP REST endpoint for the PolicyConfigurator service.
+func (pc *PolicyConfigurator) Path() string {
+	return endpoint
+}
+
+// Method returns the HTTP REST method for the configure policy service.
+func (pc *PolicyConfigurator) Method() string {
+	return http.MethodPost
+}
+
+// Handler returns the HTTP REST handle for the PolicyConfigurator service.
+func (pc *PolicyConfigurator) Handler() common.HTTPRequestHandler {
+	return pc.handle
+}
+
+// New returns a new PolicyConfigurator.
+func New(authCfg *resthandler.Config, s spi.Store, verifier signatureVerifier, cfgStore storage.Store) *PolicyConfigurator { //nolint:lll
+	h := &PolicyConfigurator{
+		VerifyActorInSignature: authCfg.VerifyActorInSignature,
+		configStore:            cfgStore,
+	}
+
+	h.AuthHandler = resthandler.NewAuthHandler(authCfg, endpoint, http.MethodPost, s, verifier, nil)
```

Comment:
Is this just a client-to-server call? If so, I'm not sure that we should be using HTTP signature verification since HTTP signatures are generally used for server-to-server invocations. I think we should just use bearer tokens, like: https://github.com/trustbloc/orb/blob/a33d9f69adc26be9396d34f5da06aa901d24ef87/cmd/orb-server/startcmd/start.go#L643.

Reply:
Done